docs/TUTORIAL.md

```diff
@@ -1,3 +1,305 @@
+            probe: {
+                path: "/healthz",
+                port: 10254,
+                initialDelaySeconds: 10,
+                timeoutSeconds: 1,
+                scheme: "HTTP",
+            },
+            ports: {
+                http: 80,
+                https: 443,
+            },
+        },
+    },
+};
+
+// define the metadata that the IngressController object needs
+local metadata = deployment.mixin.metadataType;
+local controllerMetadata = metadata.new() +
+    metadata.name(conf.ingressController.name) +
+    metadata.labels(conf.ingressController.selector) +
+    metadata.namespace(conf.namespace);
+
+// define the container for the deployment
+local nginxContainer = container.new(conf.ingressController.name, conf.ingressController.container.image) +
+    container.ports([
+        container.portsType.containerPort(conf.ingressController.container.ports.http) +
+        container.portsType.hostPort(conf.ingressController.container.ports.http),
+
+        container.portsType.containerPort(conf.ingressController.container.ports.https) +
+        container.portsType.hostPort(conf.ingressController.container.ports.https),
+    ]);
+
+// define the deployment
+local ingressController = deployment.new(conf.ingressController.name, conf.ingressController.replicas, [nginxContainer], conf.ingressController.selector) +
+    deployment.mixin.metadata.mixinInstance(controllerMetadata) +
+    deployment.mixin.spec.template.metadata.annotations(conf.ingressController.annotations) +
+    deployment.mixin.spec.template.spec.terminationGracePeriodSeconds(conf.ingressController.terminationGracePeriodSeconds);
+
+// call the deployment
+k.core.v1.list.new(ingressController)
+```
+
+Save the file as `ingressController.jsonnet` at the root of your **ksonnet** fork,
+and run:
+
+```bash
+jsonnet ingressController.jsonnet > deployment.yaml
+kubectl apply -f deployment.yaml
+```
+
+NOTE: If you save the file elsewhere, you must explicitly specify the path to the **ksonnet** 
+libraries. The command looks like: `jsonnet -J </path/to/ksonnet-lib> ingressController.jsonnet`
+
+The generated YAML looks like this:
+
+```yaml
+apiVersion: v1
+items:
+  - apiVersion: extensions/v1beta1
+    kind: Deployment
+    metadata:
+      labels:
+        run: nginx-ingress-controller
+      name: nginx-ingress-controller
+      namespace: nginx-ingress
+    spec:
+      replicas: 2
+      template:
+        metadata:
+          labels:
+            run: nginx-ingress-controller
+        spec:
+          containers:
+            - image: 'gcr.io/google_containers/nginx-ingress-controller:0.9.0-beta.7'
+              name: nginx-ingress-controller
+              ports:
+                - containerPort: 80
+                  hostPort: 80
+                - containerPort: 443
+                  hostPort: 443
+          terminationGracePeriodSeconds: 60
+kind: List
+```
+
+Now you can add other objects simply, using additional built-in mixins. The Service 
```

Comment:
This might be just personal preference, so you can take with a grain of salt, but consider having "Ingress controller" and "Service definition" subheaders to denote the separate setups?